documentation/writing_tests.md

```diff
@@ -152,28 +176,41 @@ Mocha is used by our test scripts in our `package.json` files. We typically invo
+
+To use Mocha from the `package.json`, our systems will expect to encounter two scripts, one called `unit-test` for tests that will be [executed during Pull Request validation](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L226-L233), which won't ever reach to live resources, and another called `integration-test` for our [nightly and release builds](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/pipelines/templates/jobs/archetype-sdk-integration.yml#L114), which will be expected to reach to live resources. We assume that the distinction of when to reach to what resources will be done within the tests (either from  [The Recorder](#the-recorder) or through [Using conditionals](#using-conditionals)). With this in mind, and limiting `nyc` to only run on the `integration-test` script (since `unit-test` will be used for debugging), we will end up with the following scripts:
+
+```json
+    "integration-test:node": "find dist-esm/test -name '*.spec.js' | xargs nyc mocha -r esm --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 180000 --full-trace",
+    "unit-test:node": "mocha --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 180000 --full-trace dist-test/index.node.js",
+```
+
+Keep in mind that Mocha will be directly called from our `package.json` scripts only for our **NodeJS** tests. For our browser tests, we will be using [Karma](#karma).
 
 You can look at how Mocha's configuration is present in our [template project's package.json file](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/template/template/package.json).
 
+#### Code coverage with Mocha and nyc
+
+Our `integration-test` script will output code-coverage on our nightly and release builds. It's recommended to run this manually to confirm that code coverage is high. Code coverage should be above 80%, though code coverage should not change how we write code or tests.
+
 #### Handling timeouts
 
-All of the tests in the Azure SDK repository should have a timeout. In the previous section, we have discussed how this timeout is defined for mocha, though the parameter `--timeout` with an additional milliseconds amount, which is sent during the invocation of mocha. Now we will define how to know how many milliseconds to pass to this parameter.
+All of the tests in the Azure SDK repository should have a timeout. In the previous section, we have discussed how this timeout is defined for _mocha_, though the parameter `--timeout` with an additional milliseconds amount, which is sent during the invocation of mocha. Now we will define how to know how many milliseconds to pass to this parameter.
```

Comment:
I think our pipelines themselves have a default timeout too and mocha cannot go beyond them.

Reply:
@jeremymeng was asking about this the other week so he can probably provide context on how you actually change it (or are supposed to change it).

Reply:
The build pipeline timeout is for the total time of the build.  This timeout we passed to mocha is the time out for one test (e.g., one `it()` function). I think what @sadasant has here is proper as it's taking from the slowest passing test from the history.  I don't believe we need to take the sum of running times in the worst case and make sure the sum is below build pipeline timeout limit.

Reply:
Thank you, Jeremy.

This timeout strategy is taken in coordination with Engineering.

Something to consider is that if we ever reach the engineering timeout (of one hour) all of our logs will most likely disappear. I think this might be an Engineering bug so I'm not sure if I should document this in the guidelines. The sum of our individual timeouts should still be far from reaching the Engineering's timeout. It will be easy to detect if we ever need to increase the Engineering timeout by checking the worst case scenario from all of our tests. If I'm not mistaken, I believe that our worse case scenario, when it passes, is at about 50% of the current Engineering timeout.